genai-perf/genai_perf/inputs/converters/base_converter.py

```diff
@@ -25,18 +25,19 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import random
-from typing import Dict, List, Union, cast
+from typing import Dict, List, Union
 
 from genai_perf.exceptions import GenAIPerfException
 from genai_perf.inputs.input_constants import ModelSelectionStrategy
 from genai_perf.inputs.inputs_config import InputsConfig
+from genai_perf.inputs.retrievers.generic_dataset import GenericDataset
 
 
 class BaseConverter:
 
     _CONTENT_NAMES: List[str]
 
-    def convert(self, generic_dataset: Dict, config: InputsConfig) -> Dict:
+    def convert(self, generic_dataset: GenericDataset, config: InputsConfig) -> Dict:
```

Comment:
Just a general boy scout comment. It's good practice to not have naked Dict/List as types. If possible try to clean this up when you touch lines like this.

Reply:
The output of this file is the whole dictionary with several nested data structures that we turn into a JSON. The dictionary schema is going to be convert-specific.

GenericDataset is supposed to make the earlier steps easier, but the data structure itself here is a very complicated schema. After this work does a lot of cleanup, it's possible that we could talk about if there's a better way to type hint for this. I was just updating the one relevant type hint for all the converters, but it's possible that made it look like I was doing full clean-up throughout. That wasn't the plan. That said, if you have ideas for how this could be type hinted, I'd be happy to open a ticket to do so. Ideally in a future PR, so that this PR doesn't blow up with all of the work for this feature and related cleanup.

Reply:
I would just create a type `Dict[any, any]` and make that the output of the function. It least it shows that you've thought about it and this is intended.

Reply:
Thanks for the suggestion! We'll add that.

CC: @debermudez for awareness

Reply:
this is exactly what we were thinking. thanks for the confirmation and call out @nv-braf